wrapper/src/main/java/software/amazon/jdbc/targetdriverdialect/PgTargetDriverDialect.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.targetdriverdialect;
+
+import java.sql.Driver;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+import javax.sql.DataSource;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import software.amazon.jdbc.HostSpec;
+
+public class PgTargetDriverDialect extends GenericTargetDriverDialect {
+
+  private static final Set<String> dataSourceClassMap = new HashSet<>(Arrays.asList(
+          "org.postgresql.ds.PGSimpleDataSource",
+          "org.postgresql.ds.PGPoolingDataSource",
+          "org.postgresql.ds.PGConnectionPoolDataSource"));
+
+  @Override
+  public boolean isDialect(Driver driver) {
+    return "org.postgresql.Driver".equals(driver.getClass().getName());
+  }
+
+  @Override
+  public boolean isDialect(String dataSourceClass) {
+    return dataSourceClassMap.contains(dataSourceClass);
+  }
+
+  @Override
+  public void prepareDataSource(
+      final @NonNull DataSource dataSource,
+      final @NonNull String protocol,
+      final @NonNull HostSpec hostSpec,
+      final @NonNull Properties props,
+      final @Nullable String serverPropertyName,
+      final @Nullable String portPropertyName,
+      final @Nullable String urlPropertyName,
+      final @Nullable String databasePropertyName) throws SQLException {
+
+    // The logic is isolated to a separated class since it uses
+    // direct reference to org.postgresql.ds.common.BaseDataSource
```

Comment:
Referring to the code comments above, is the idea to decouple the Dialect class from `org.postgresql.ds.common.BaseDataSource`?

Reply:
That's correct. The idea is to decouple different dialect classes. Since `PgTargetDriverDialect` is instantiated in `TargetDriverDialectManager`, it's impossible to use any classes from `org.postgresql` package in `PgTargetDriverDialect`. In this case AWS JDBC Driver would have a dependency on `org.postgresql` package. That's why all logic is isolated in `PgDataSourceHelper` class. This helper class has direct references to `org.postgresql` package but it's instantiated when it's needed only, i.e. when the target driver is actually `pgjdbc` driver.

Reply:
Thank you for the explanation!